src/lib/viewers/doc/BoxExcel/sheet.js

```diff
@@ -0,0 +1,640 @@
+        return sheet[pos];
+    };
+
+    /**
+     * Helper function of _getBorder
+     *
+     * @param  {Object} style style property in a cell object
+     * @return {string}       hex color string
+     */
+    _getBorderColor = (style) => {
+        return style && style.color ? _parseColor(style.color) : c.gridGrey;
+    };
+
+    /**
+     * Helper function of _getBorder
+     *
+     * @param  {style} style style property in a cell object
+     * @return {string}      border width string, default: '1px'
+     */
+    _getBorderWidth = (style) => {
+        return style && style.style && borderWidthMap[style.style] ? borderWidthMap[style.style] : '1px';
+    };
+
+    /**
+     * Get border style, including borderColor and borderWidth
+     *
+     * @param  {Object} style style property in a cell object
+     * @return {Object}       React style object
+     */
+    _getBorder = (style) => {
+        const borderColor = `${this._getBorderColor(style.top)} ${this._getBorderColor(
+            style.right
+        )} ${this._getBorderColor(style.bottom)} ${this._getBorderColor(style.left)}`;
+        const borderWidth = `${style.top ? this._getBorderWidth(style.top) : '0px'} ${
+            style.right ? this._getBorderWidth(style.right) : '1px'
+        } ${style.bottom ? this._getBorderWidth(style.bottom) : '1px'} ${
+            style.left ? this._getBorderWidth(style.left) : '0px'
+        }`;
+        return { borderColor, borderWidth };
+    };
+
+    /**
+     * Get cell content
+     *
+     * @param  {number} rowIndex    row index, 1 basis, 0 is header
+     * @param  {number} columnIndex col index, 1 basis, 0 is header
+     * @return {string}             cell content
+     */
+    _getCellContent = (rowIndex, columnIndex) => {
+        if (rowIndex === 0 && columnIndex === 0) {
+            return '';
+        } else if (rowIndex === 0) {
+            /* global XLSX */
+            return XLSX.utils.encode_col(columnIndex - 1);
+        } else if (columnIndex === 0) {
+            return rowIndex;
+        }
+        const cell = this._getCell(rowIndex, columnIndex);
+        if (!cell) return '';
+        if (cell.t === 'd') {
+            return cell.w ? cell.w : dateConvertor(cell.v, cell.z);
+        }
+        return cell.w;
+    };
+
+    /**
+     * Render single cell for at (rowIndex, columnIndex)
+     *
+     * @param  {any} key               cell key
+     * @param  {number} columnIndex    column index 1 basis, 0 means header
+     * @param  {number} rowIndex       row index 1 basis, 0 means header
+     * @param  {number} scrollToColumn focused col index, 1 basis
+     * @param  {number} scrollToRow    focused row index, 1 basis
+     * @param  {Object} style          React style object
+     * @return {jsx}                   Rendered jsx
+     */
+    _cellRenderer = ({ key, columnIndex, rowIndex, scrollToColumn, scrollToRow, style }) => {
+        const { merges, gridlines, zoomBase } = this.state;
```

Comment:
Move the cell style parsing into it's own method and maybe update the state which can then be used by the cellRender() method